### ISSUE TITLE
Substitute version only for Source0

### ISSUE THREAD

**Issue by phracek** [rebase-helper/rebase-helper#145]
During testing upstream monitoring service I discovered that SourceX (except Source0) can have also {version} tag. Like here for rubygem-web-console

```
$ head -n 20 rubygem-web-console.spec 
# Generated from web-console-2.0.0.gem by gem2rpm -*- rpm-spec -*-
%global gem_name web-console

Name: rubygem-%{gem_name}
Version: 2.2.1
Release: 1%{?dist}
Summary: A debugging tool for your Ruby on Rails applications
Group: Development/Languages
License: MIT
URL: https://github.com/rails/web-console
Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
# git clone https://github.com/rails/web-console.git && cd web-console
# git checkout v2.2.1 && tar czvf web-console-2.2.1-tests.tgz test/
Source1: %{gem_name}-%{version}-tests.tgz
BuildRequires: ruby(release)
$ 
```

I would suggest to translate only Source0. What do you think about it?
Comments are more then welcome.

**Comment by nforro** [comment]
I think there is no other solution than substituting the %{version} macro with old version string, if the source isn't downloadable or can't be downloaded successfully. But there should definitely be comment about it in the spec file.

**Comment by thozza** [comment]
The approach with keeping the old version for Source1 is questionable. Since in this example these are tests, it is highly possible that running tests during the build will fail. But I agree that it may be the only sane thing to try out and inform the user about this. Note that there is a way to construct the tests in comments above the Source1. It would be better to take these into consideration and try to construct the tarball by ourself.

**Comment by junaruga** [comment]
This `rubygem-web-console` is maybe my package.
This pattern is popular for rubygem packages.

This code intends to create Source1 manually right now.
For example, we have to prepare `web-console-2.2.2-tests.tgz` manually to prepare Source 1 to update to version 2.2.2

```
# git clone https://github.com/rails/web-console.git && cd web-console
# git checkout v2.2.2 && tar czvf web-console-2.2.2-tests.tgz test/
```

But ideally, below looks good for automation.

```
Detect a pattern in a spec file => Act something to edit the spec file.
```

**Comment by skisela** [comment]
This was resolved by #327
